### PR TITLE
Updated decode and encode functions to avoid forced downcasting

### PR DESCRIPTION
**Pull request by oscardemoya**

### CODE REVIEW

**Comment by kean** [comment]
Is there a problem with force cast?

**Comment by oscardemoya** [comment]
> Is there a problem you encountered with force cast?

Not really, just to prevent any crash if by mistake some change is done like implementing `OptionalDecoding` to other type different than Optional.

File: Sources/Get/DataLoader.swift [hunk 1]
```diff
@@ -337,30 +337,31 @@ extension Optional: OptionalDecoding {}
...
-    if data.isEmpty, T.self is OptionalDecoding.Type {
-        return Optional<Decodable>.none as! T
-    } else if T.self == Data.self {
-        return data as! T
-    } else if T.self == String.self {
-        guard let string = String(data: data, encoding: .utf8) else {
+    if data.isEmpty, T.self is OptionalDecoding.Type, let value = Optional<Decodable>.none as? T {
+        return value
+    }
+    if T.self == Data.self, let value = data as? T {
```
Comment (grsouza) [inline Sources/Get/DataLoader.swift:353]: I don't think checking for the type first is needed anymore, `let value = data as? T` should do the work already.

**Comment by kean** [comment]
Thanks for the PR, @oscardemoya. I understand the reasoning behind the changes, but since it's mainly a cosmetic change, I think I'd rather keep the original version – it's a bit more clear to me at least.

**Pull request closed without merging**